src/appleseed.shaders/src/appleseed/as_noise2d.osl

```diff
@@ -364,166 +381,196 @@ shader as_noise2d
-        }
-        else if (noisetype == "voronoise")
-        {
-            signal = voronoise2d(
-                coords[0],
-                coords[1],
-                in_jittering,
-                in_smoothness) * 2.0 - 1.0;
-        }
-        else if (noisetype == "uvalue")
-        {
-            signal = value_noise2d(coords[0], coords[1]);
-        }
-        else if (noisetype == "value")
-        {
-            signal = value_noise2d(coords[0], coords[1]) * 2.0 - 1.0;
-        }
-        else if (in_periodic)
-        {
-            signal = (float) pnoise(
-                noisetype,
-                coords[0],
-                coords[1],
-                in_period_x,
-                in_period_y);
-        }
-        else
-        {
-            signal = (float) noise(noisetype, coords[0], coords[1]);
-        }
+            if (noisetype == "gabor")
+            {
+                signal = (float) noise(
+                    "gabor",
+                    coords[0],
+                    coords[1],
+                    "anisotropic", in_anisotropy,
+                    "direction", in_direction,
+                    "bandwidth", in_bandwidth,
+                    "impulses", in_impulses,
+                    "do_filter", in_filter_noise) * 0.5 + 0.5;
+            }
+            else if (noisetype == "voronoise")
+            {
+                signal = voronoise2d(
+                    coords[0],
+                    coords[1],
+                    in_jittering,
+                    in_smoothness);
+            }
+            else if (noisetype == "value")
+            {
+                signal = value_noise2d(coords[0], coords[1]);
+            }
+            else if (in_periodic)
+            {
+                signal = (float) pnoise(
+                    noisetype,
+                    coords[0],
+                    coords[1],
+                    in_period_x,
+                    in_period_y);
+            }
+            else
+            {
+                signal = (float) noise(
+                    noisetype, coords[0], coords[1]);
+            }
 
-        if (in_inflection)
-        {
-            signal = abs(signal);
-        }
-        
-        if (in_ridges)
-        {
-            signal = in_offset - signal;
-            signal *= signal;
-            signal *= weight;
-            weight = clamp(signal * 2.0, 0.0, 1.0);
-        }
+            if (in_signed_noise)
+            {
+                signal = signal * 2.0 - 1.0;
+            }
 
-        out_color += amplitude * signal;
+            if (in_inflection)
+            {
+                signal = abs(signal);
+            }
 
-        amplitude *= noise_gain;
-        new_st[0] *= lacunarity;
-        new_st[1] *= lacunarity;
+            if (in_ridges)
+            {
+                signal = in_offset - signal;
+                signal *= signal;
+                signal *= weight;
+                weight = clamp(signal * 2.0, 0.0, 1.0);
```

Comment:
Is assigning to `weight` really what you intended to do here?

Also, unless `weight` is negative, `signal` cannot be negative since you square it.

Reply:
The assignment yes, we start with 1.0 in the first iteration, then store the signal value into weight for the next octave. You're right about the clamp, i wasn't sure which of min() or clamp() would be faster.